x-pack/plugins/canvas/public/apps/workpad/routes.ts

```diff
@@ -70,8 +70,8 @@ export const routes = [
               const fetchedWorkpad = await workpadService.get(params.id);
 
               const { assets, ...workpad } = fetchedWorkpad;
-              dispatch(setWorkpad(workpad));
               dispatch(setAssets(assets));
+              dispatch(setWorkpad(workpad));
```

Comment:
Without swapping those lines, the asset functions in the workspace weren't working correctly on the initial loading. 
That was caused by the change introduced in this PR. Previously we were using promises, and the expression invocation was happening on the next tick. But after switching to the observables, the expressions are invoked on the current tick due to the synchronous nature of the observables.
There was no difference previously for the canvas' code because the assets were initialized before the next tick, but after changing the expressions, the order matters.